ocs_ci/ocs/longevity.py

```diff
@@ -555,6 +558,255 @@ def stage_0(
+    Function to handle automation of Longevity Stage 4 i.e.
+        1. Creation / Deletion of PODs, PVCs of different types + fill data upto 25% of mount point space.
+        2. Creation / Deletion of Clones of the given PVCs.
+        3. Creation / Deletion of VolumeSnapshots of the given PVCs.
+        4. Restore the created VolumeSnapshots into a new set of PVCs.
+        5. Expansion of size of the original PVCs.
+
+
+    Args:
+        project_factory : Fixture to create a new Project.
+        multi_pvc_factory : Fixture to create multiple PVCs of different access modes and interface types.
+        pod_factory : Fixture to create new PODs.
+        multi_pvc_clone_factory : Fixture to create a clone from each PVC in the provided list of PVCs.
+        multi_snapshot_factory : Fixture to create a VolumeSnapshot of each PVC in the provided list of PVCs.
+        multi_snapshot_restore_factory : Fixture to create a set of new PVCs out of each VolumeSnapshot provided in the
+                                            list.
+        teardown_factory : Fixture to tear down a resource that was created during the test.
+        num_of_pvcs (int) : Total Number of PVCs we want to create.
+        pvc_size (int) : Size of each PVC in GB.
+        run_time (int) : Total Run Time in minutes.
+        pvc_size_new (int) : Size of the expanded PVC in GB.
+
+    """
+    end_time = datetime.now() + timedelta(minutes=run_time)
+    cycle_no = 0
+
+    while datetime.now() < end_time:
+        cycle_no += 1
+        log.info(f"------------STARTING CYCLE:{cycle_no}------------")
+
+        namespace = "stage-4-cycle-" + str(cycle_no)
+        project = project_factory(project_name=namespace)
+        executor = ThreadPoolExecutor(max_workers=1)
+        pvc_objs = list()
+
+        for interface in (constants.CEPHFILESYSTEM, constants.CEPHBLOCKPOOL):
+            if interface == constants.CEPHFILESYSTEM:
+                access_modes = [constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX]
+                num_of_pvc = num_of_pvcs // 2
+            else:
+                access_modes = [
+                    constants.ACCESS_MODE_RWO,
+                    constants.ACCESS_MODE_RWO + "-" + constants.VOLUME_MODE_BLOCK,
+                    constants.ACCESS_MODE_RWX + "-" + constants.VOLUME_MODE_BLOCK,
+                ]
+                num_of_pvc = num_of_pvcs - num_of_pvcs // 2
+
+            # Create PVCs
+            if num_of_pvc > 0:
+                pvc_objs_tmp = multi_pvc_factory(
+                    interface=interface,
+                    size=pvc_size,
+                    project=project,
+                    access_modes=access_modes,
+                    status=constants.STATUS_BOUND,
+                    num_of_pvc=num_of_pvc,
+                    wait_each=True,
+                )
+                log.info("PVC creation was successful.")
+                pvc_objs.extend(pvc_objs_tmp)
+
+            else:
+                log.info(
```

Comment:
Should we log it as an error? and do we need to throw exception?

Reply:
Ack, will log it is as an error.